Code/Framework/AzCore/AzCore/Debug/PerformanceCollector.cpp

```diff
@@ -0,0 +1,253 @@
+        m_statisticsManager.GetAllStatistics(statistics);
+        m_periodicEventStamps.clear();
+        for (auto statistic : statistics)
+        {
+            m_periodicEventStamps[statistic->GetName()] = {};
+        }
+    }
+
+    void PerformanceCollector::UpdateDataLogType(DataLogType newValue)
+    {
+        if (m_dataLogType == newValue)
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_dataLogType = newValue;
+    }
+
+    void PerformanceCollector::UpdateFrameCountPerCaptureBatch(AZ::u32 newValue)
+    {
+        if (m_frameCountPerCaptureBatch == newValue)
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_frameCountPerCaptureBatch = newValue;
+    }
+
+    void PerformanceCollector::UpdateWaitTimeBeforeEachBatch(AZStd::chrono::seconds seconds)
+    {
+        if (m_waitTimeBeforeEachBatch.count() == seconds.count())
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_waitTimeBeforeEachBatch = seconds;
+    }
+
+    void PerformanceCollector::UpdateNumberOfCaptureBatches(AZ::u32 newValue)
+    {
+        if (m_numberOfCaptureBatches == newValue)
+        {
+            return;
+        }
+        if (m_numberOfCaptureBatches == 0)
+        {
+            CreateOutputJsonFile();
+        }
+        m_numberOfCaptureBatches = newValue;
+        AZ_TracePrintf(LogName, "%s updated value to %u", __FUNCTION__, m_numberOfCaptureBatches);
+    }
+
+    void PerformanceCollector::CreateOutputJsonFile()
+    {
+        AZStd::chrono::steady_clock::time_point now = AZStd::chrono::steady_clock::now();
+
+        m_outputFilePath.clear();
+        m_outputDataBuffer.clear();
+
+        AZ::SettingsRegistryInterface* settingsRegistry = AZ::SettingsRegistry::Get();
+        if (!settingsRegistry)
+        {
+            // Most likely running under unit test.
+            auto bufferStream = AZStd::make_unique<AZ::IO::ByteContainerStream<AZStd::string>>(&m_outputDataBuffer);
+            m_eventLogger.ResetStream(AZStd::move(bufferStream));
+        }
+        else
+        {
+            settingsRegistry->Get(m_outputFilePath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectUserPath);
+            m_outputFilePath /= AZStd::string::format("Performance_%s_%lld.json", m_logCategory.c_str(), now.time_since_epoch().count());
```

Comment:
I added set of [DateFormat](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Date/DateFormat.h#L508-L512) functions that can be used to get a filename sanitized timestamp.

You can invoke `GetFilenameCompatibleFormatNowWithMicroseconds` to populate an `AZ::Data::Iso8601TimestampString` string with a filename compatible string

Reply:
Nice! The code is now using `GetFilenameCompatibleFormatNowWithMicroseconds`.

Reply:
Hmmm, this function is not defined: GetFilenameCompatibleFormatNowWithMicroseconds(). I'm getting linker errors and when I grep for it in the code base it yields 0 files.

Reply:
Changed to this instead:
```
            auto now = AZStd::chrono::utc_clock::now();
            AZ::date::Iso8601TimestampString utcTimestamp;
            AZ::date::GetFilenameCompatibleFormatWithMicroseconds(utcTimestamp, now);
```

Reply:
I am adding a fixed for this in #12560